pkg/virt-launcher/virtwrap/converter/network.go

```diff
@@ -120,6 +120,18 @@ func createDomainInterfaces(vmi *v1.VirtualMachineInstance, domain *api.Domain,
 				domainIface.Rom = &api.Rom{Enabled: "no"}
 			}
 		}
+
+		if ifaceType == "virtio" && c.UseLaunchSecurity {
+			if domainIface.Rom != nil && domainIface.Rom.Enabled != "no" {
+				return nil, fmt.Errorf("%s: Rom.Enabled = '%s' not allowed with launch security", iface.Name, domainIface.Rom.Enabled)
+			}
+			domainIface.Rom = &api.Rom{Enabled: "no"}
+			if domainIface.Driver != nil {
+				domainIface.Driver.IOMMU = "on"
+			} else {
+				domainIface.Driver = &api.InterfaceDriver{Name: "vhost", IOMMU: "on"}
+			}
+		}
```

Comment:
TBH I was not able to find the flow when `Rom.Enabled` is set to `yes`. So this is mostly a precaution measure for the case when someone changes the network configuration code. Besides, the `Rom` element is not exposed in the VMI spec and moving the check to the validation webhook will require similar logic as in the converter. Does it seem reasonable?

Reply:
This happens to me regularly (not finding the flow). Usually, it's because we just pass the value to libvirt, where the logic actually is.
It's the case here. See https://github.com/kubevirt/kubevirt/blob/403f1a864e58cc5db9774bf7fe5880e281d3d638/pkg/virt-launcher/virtwrap/api/schema.go#L708
Rom defaults to yes in libvirt, we just set it to no when the user didn't define a boot order for the interface, to disable iPXE.
Your code here probably needs reworking to take into account that the Rom attribute is true by default.

Reply:
Yeah, I somehow did not take into account the defaults from libvirt. Valid point. Will look at how to rework the code here. @jean-edouard , thanks for the info :+1:

Reply:
Updated the code to handle the defaults. Now Rom is explicitly disabled if SEV is requested.

Reply:
I think that approach is fine, but only if no `bootOrder` was specified for the interface, and if `rom` was not manually enabled.
Otherwise, that would mean we silently override parameters explicitly set by the user, which doesn't sound good...
I'd suggest keeping this code as is, but in the admitter just reject VM(I)s that request bootable NICs + SEV.
How does that sound?

Reply:
I was rather thinking about documenting that with SEV Rom will be disabled. But maybe you are right: having a validation in the admitter might be a better option.